File: MATLAB/save_read_data_txt.m
% Matlab R2022a
% Keywords:
%    save or read data in txt
%    copy strings and arrays
%    format of read and print
% By Edward Z. Wu 2023-02-10

%% read

folder = 'E:\MATLABproject\';
% txtformat="%.8f\n"; % .8f won't work in reading.
txtformat="%f\n";
fpath=[folder 'sig_read.txt'];
fileID = fopen(fpath,'r');
sig_read=fscanf(fileID,txtformat);
fclose(fileID);

%% save

sig_write=sig_read;

txtformat="%.8f\n";
fpath=[folder 'sig_write.txt'];
fileID = fopen(fpath,'w');
fprintf(fileID,txtformat,sig_write);
fclose(fileID);

File: Python/save_read_data_txt.py
import os
import copy

# 'r'	It opens a file for reading only.
# 'w'	It opens a file for writing. If the file exists, it overwrites it, otherwise, it creates a new file.
# 'a'	It opens a file for appending only. If the file doesn’t exist, it creates the file.
# 'x'	It creates a new file. If the file exists, it fails.
# '+'	It opens a file for updating.
# 't'   Default mode
# 'b'   Binary mode


folder='E:\\Pythonproject\\LearningNotes\\'
fname='sig_read.txt'

fpath=folder+fname
f = open(fpath, 'r')
#data1 = f.read().splitlines()
data1 = f.read() # "1.xxx\n0.xxx\n...." read all data in a string
data2 = f.read()
# after read() or readline(), f has already been read, return empty, so data2=''
f.close()

with open(fpath,'r') as f:
    data2 = f.read().splitlines() # read number in list[500]
    f.close()

with open(fpath,'r') as f:
    data3 = f.readlines() # read number in list[500]
    f.close()
